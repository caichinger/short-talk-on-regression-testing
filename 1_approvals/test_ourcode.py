"""
Regression tests.
"""
from functools import singledispatch

import pandas as pd
from approvaltests.approvals import verify


@singledispatch
def ourverfiy(obj):
    verify(obj)


@ourverfiy.register(pd.DataFrame)
def _(obj):
    return verify(obj.to_json(orient="split", indent=4))


def test_thing():
    # prepare
    # act
    result = "Very important! =)"
    # assert
    ourverfiy(result)


def test_with_dataframe():
    result = pd.DataFrame({"foo": [1, 2], "bar": [10, 20]})
    ourverfiy(result)
