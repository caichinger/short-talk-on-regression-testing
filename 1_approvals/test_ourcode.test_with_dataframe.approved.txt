{
    "columns":[
        "foo",
        "bar"
    ],
    "index":[
        0,
        1
    ],
    "data":[
        [
            1,
            10
        ],
        [
            2,
            20
        ]
    ]
}
